llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2416,6 +2416,28 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpLoad>(SPIRVValue *const s
         (void(loadType)); // unused
         return ConstantVector::get({m_builder->getInt32(0), m_builder->getInt32(0)});
       }
+#if GPURT_CLIENT_INTERFACE_MAJOR_VERSION >= 34
+      else {
+        // From now on, AS header may start at a non-zero offset, GPURT now request base offset of the resource, and it
+        // will calculate the actual GPUVA, instead of compiler providing one loaded from offset 0.
+        unsigned binding = 0u;
+        unsigned descSet = 0u;
+        bool valid = spvLoad->getSrc()->hasDecorate(DecorationBinding, 0, &binding);
+        valid &= spvLoad->getSrc()->hasDecorate(DecorationDescriptorSet, 0, &descSet);
+        assert(valid);
+
+        auto descTy = ResourceNodeType::DescriptorBuffer;
+        auto descPtr = getBuilder()->CreateGetDescPtr(descTy, descTy, descSet, binding);
+        auto desc = getBuilder()->CreateLoad(getBuilder()->getDescTy(descTy), descPtr);
```

Comment:
Since we're only using the first 64 bit, it's better to explicitly load a `<2 x i32>`. Then you can extract, fixup, and re-insert the element index 1.